packages/apollo-server-plugin-base/src/index.ts

```diff
@@ -24,50 +47,29 @@ export interface ApolloServerPlugin<TContext extends Record<string, any> = Recor
 
```

Comment:
I can't leave a comment any higher than here, but do we want a `type GraphQLRequestContextRequestDidStart` as well for L44 and elsewhere? I think something like that could be used here as well: https://github.com/apollographql/apollo-server/pull/3902/files#diff-2b2e6074dd019117405fb5ba4938b17eR215-R218

Reply:
Possible.  The permalink you linked to isn't working, however, so I can't be certain.

Generally though, I think that `GraphQLRequestContext<TContext>` is a fairly concise type (and name) for `requestContext` (which is the _request context_?).

Anyhow, I'm curious what that link was to.  I do agree we can spread this more widely, as appropriate (though I did avoid `graphql-extensions` due to its imminent demise.)

Reply:
Here's the permalink you deserved!
https://github.com/apollographql/apollo-server/blob/c177acd7959aaf516f8ac814f1a6bb2e6d3ed2e6/packages/apollo-server-plugin-base/src/index.ts#L21

Currently L44 in your PR, however.

Safe to say we've gone and replaced most of the `GraphQLRequestContext`s, this is just one that wasn't replaced as the others were.

Reply:
Ah, thanks!

Not replacing this one was intentional since the other life-cycles sub-sets of this type and all live (nested) within `GraphQLRequestContext`.  Are you suggesting that we alias `GraphQLRequestContextRequestWillStart<TContext>` to `GraphQLRequestContext<TContext>` to help with clarity?  They absolutely must be identical at this stage of the pipeline; this is the base type that all other later phases are based on.

Reply:
Ah, that makes sense. I think my original intention was to be able to reuse this type elsewhere. Possibly here https://git.io/JvSaQ? Though I no longer see the value in that given that it's `extensionStack` code. As an aside, I don't believe the types between the extensions and plugin here are compatible - the plugin `requestDidStart` doesn't have the same `WithRequired` that extensions do.

I think this is good to leave as is unless you feel aliasing for the sake of clarity would be beneficial.

Reply:
Correct, the two pipelines currently have different contracts as to what properties are guaranteed, though for no clear reason.  I basically left anything alone that was touching extensions. 😄